libs/mocks/src/connectors_gateway_routers.rs

```diff
@@ -1,33 +1,85 @@
 use cfg_traits::connectors::Router;
-use codec::{Decode, Encode, MaxEncodedLen};
-use frame_support::dispatch::DispatchResult;
-use scale_info::TypeInfo;
-use sp_std::{default::Default, marker::PhantomData};
+use frame_support::{dispatch::DispatchResult, pallet_prelude::*};
+use mock_builder::{execute_call, register_call};
+use sp_std::default::Default;
 
 use crate::MessageMock;
 
-#[derive(Default, Debug, Encode, Decode, Clone, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
-pub struct DomainRouterMock<T: frame_system::Config> {
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::storage]
+	pub(super) type CallIds<T: Config> = StorageMap<
+		_,
+		Blake2_128Concat,
+		<Blake2_128 as frame_support::StorageHasher>::Output,
+		mock_builder::CallId,
+	>;
+
+	impl<T: Config> Pallet<T> {
+		pub fn mock_init(f: impl Fn() -> DispatchResult + 'static) {
+			register_call!(move |()| f());
+		}
+
+		pub fn mock_send(f: impl Fn(T::AccountId, MessageMock) -> DispatchResult + 'static) {
+			register_call!(move |(sender, message)| f(sender, message));
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
```

Comment:
If you use the impl trait here:
```rust
impl<T: Config> Router for Pallet<T> {
```

I think you do not need `RouterMock` wrapper

Reply:
It's what I initially did but then I get loads of errors when trying to store the pallet ^^

This eventually ends up in the storage and it means that it has to implement `Encode, Decode, MaxEncodedLen, TypeInfo` etc.

Reply:
Mmm... let me think. All these use-cases/experience can be used to improve `mock-builder` for other use cases 👍🏻